packages/cozy-harvest-lib/src/components/Popup.jsx

```diff
@@ -87,15 +87,33 @@ export class Popup extends PureComponent {
 
   showPopup() {
     const { height, width, title, url } = this.props
-    const popup = popupCenter(url, title, width, height)
+    const { w, h, top, left } = popupCenter(width, height)
+    /**
+     * ATM we also use window.open on Native App in order to open
+     * InAppBrowser. But some provider (Google for instance) will
+     * block us. We need to use a SafariViewController or Chrome Custom Tab.
+     * So
+     */
+    const popup = window.open(
+      url,
+      title,
+      `scrollbars=yes, width=${w}, height=${h}, top=${top}, left=${left}`
+    )
+    // Puts focus on the newWindow
+    if (popup.focus) {
+      popup.focus()
+    }
+
     this.addListeners(popup)
     this.startMonitoringClosing(popup)
     this.setState({ popup })
   }
 
   killPopup() {
-    this.removeListeners()
+    const { popup } = this.state
+    this.removeListeners(popup)
     this.stopMonitoringClosing()
+    if (!popup.closed) popup.close()
```

Comment:
Are you sure about this line? I'm not sur we'll call the killPopup

Reply:
See my comment above about triggering `Popup.componentWillUnmount`. It's working on web and I guess it worth giving it a try on mobile